### PR TITLE
TIP-838: Move user classes to security bundle or the right bundle

### PR DESCRIPTION
**Pull request by arnolanglade**
Some properties have been added to the user model, they are from the EE. We won't use in the CE but it will simply the EE that is why I did that. 

We added a system which allow to easily extend the user form type. You only need to create a new subscriber and register it in the root user form type. 

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added integration tests           | -
| Changelog updated                 | yes
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Pim/Component/User/EntityUploadedImageInterface.php [hunk 1]
```diff
@@ -41,4 +42,44 @@ public function getImageFile();
      * @return string
      */
     public function getUploadDir();
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getAssetDelayReminder();
```
Comment (jjanvier) [inline src/Pim/Component/User/EntityUploadedImageInterface.php]: I don't understand why it's here?
should be in the UserInterface no?

File: src/Pim/Component/User/Model/UserInterface.php [hunk 2]
```diff
@@ -540,4 +540,61 @@ public function getTimezone(): string;
...
+
+    /**
+     * @param CategoryInterface $defaultAssetTree
+     *
+     * @return UserInterface
+     */
+    public function setDefaultAssetTree(CategoryInterface $defaultAssetTree);
+
+    /**
+     * Does the user want to be notified of new proposals to review ?
```
Comment (ahocquard) [inline src/Pim/Component/User/Model/UserInterface.php]: `review?` (no space in english, I do quite often the mistake)

File: src/Pim/Component/User/Model/UserInterface.php [hunk 3]
```diff
@@ -540,4 +540,61 @@ public function getTimezone(): string;
...
+     * Set whether the user wants to be notified of new proposals.
+     *
+     * @param bool $proposalsToReviewNotification
+     *
+     * @return UserInterface
+     */
+    public function setProposalsToReviewNotification($proposalsToReviewNotification);
+
+    /**
+     * Does the user want to be notified when its proposals are accepted or rejected ?
```
Comment (ahocquard) [inline src/Pim/Component/User/Model/UserInterface.php]: "his proposals"

File: src/Pim/Component/User/EntityUploadedImageInterface.php (hunk 1 above)
Comment (aRn0D) [reply src/Pim/Component/User/EntityUploadedImageInterface.php]: yes you're right

File: src/Pim/Component/User/Model/User.php [hunk 4]
```diff
@@ -1066,4 +1078,76 @@ public function setTimezone(string $timezone): UserInterface
...
+     */
+    public function getAssetDelayReminder()
+    {
+        return $this->assetDelayReminder;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setAssetDelayReminder($assetDelayReminder)
```
[the remark this answers is not in the document]
Comment (aRn0D) [reply src/Pim/Component/User/Model/User.php]: I don't understand

File: src/Pim/Bundle/UserBundle/Resources/config/form_types.yml [hunk 5]
```diff
@@ -21,15 +21,20 @@ services:
...
             - '@request_stack'
-            - '@pim_user.form.subscriber.user_preferences'
             - '@pim_user.repository.role'
             - '@pim_user.repository.group'
             - '@event_dispatcher'
             - '%pim_enrich.form.type.product_grid_filter_choice.class%'
         tags:
             - { name: form.type, alias: pim_user_user }
+        calls:
+            - method: 'addEventSubscribers'
```
[the remark this answers is not in the document]
Comment (aRn0D) [reply src/Pim/Bundle/UserBundle/Resources/config/form_types.yml]: It easier to add several subscriber from different compiler pass in EE.

File: src/Pim/Component/User/Model/UserInterface.php (hunk 2 above)
Comment (aRn0D) [reply src/Pim/Component/User/Model/UserInterface.php]: copy/paste from the original code :)

File: src/Pim/Component/User/Model/UserInterface.php (hunk 3 above)
Comment (aRn0D) [reply src/Pim/Component/User/Model/UserInterface.php]: idem

File: src/Pim/Component/User/Model/User.php (hunk 4 above)
[the remark this answers is not in the document]
Comment (ahocquard) [reply src/Pim/Component/User/Model/User.php]: Seen in MP: it's only moving code, not refactoring. 👍

**Pull request merged**